src/evidently/EvidentlyManager.ts

```diff
@@ -0,0 +1,273 @@
+            return Promise.reject(
+                `Evaluations for features (${notLoadedFeatures.join(
+                    ', '
+                )}) not loaded`
+            );
+        }
+
+        // If we have evaluations in cache, return them
+        const allCachedEvaluations = this.getCachedEvaluations();
+        // If every feature in features is in allCachedEvaluations then filter allCachedEvaluations to only have those features
+        const requestedEvaluations = Object.fromEntries(
+            features
+                .map((feature) => {
+                    const evaluation = allCachedEvaluations[feature];
+                    if (evaluation) {
+                        return [feature, evaluation];
+                    } else {
+                        return undefined;
+                    }
+                })
+                .filter((entry) => entry) as [string, EvaluationResult][]
+        );
+        if (features.length === Object.keys(requestedEvaluations).length) {
+            this.addEvaluationToEvents(requestedEvaluations);
+            return Promise.resolve(requestedEvaluations);
+        }
+
+        // Else:
+        // If we are currently fetching the given features, return promise with results
+        // Otherwise give error
+        return this.lastPromise.then((evals: EvaluationResults) => {
+            const responseEvaluations = Object.fromEntries(
+                features.map((feature) => {
+                    const evaluation =
+                        evals[feature] || allCachedEvaluations[feature];
+                    if (evaluation) {
+                        return [feature, evaluation];
+                    } else {
+                        throw Error(
+                            `Evaluations for feature (${feature}) was not loaded`
+                        );
+                    }
+                })
+            );
+            this.addEvaluationToEvents(responseEvaluations);
+            return responseEvaluations;
+        });
+    }
+
+    private getCachedEvaluations(): EvaluationResults {
+        if (
+            this.useSessionStorage() &&
+            Object.keys(this.cachedEvaluations).length === 0
+        ) {
+            try {
+                const jsonEvals = sessionStorage.getItem(EVALS_CACHE_KEY);
+                const sessionStorageEvals = JSON.parse(
+                    jsonEvals as string
+                ) as EvaluationResults;
+                if (sessionStorageEvals) {
+                    this.cachedEvaluations = sessionStorageEvals;
+                }
+            } catch (e) {
+                // Ignore
+            }
+        }
+        return this.cachedEvaluations;
+    }
+
+    private addToCachedEvaluations(evaluations: EvaluationResults) {
+        Object.entries(evaluations).forEach(([featureId, evaluation]) => {
+            // Only add new evaluations to the cache
+            if (!this.cachedEvaluations.hasOwnProperty(featureId)) {
```

Comment:
question: Is this check necessary?

Reply:
There are some rare edge cases where the project configuration could change between evaluation loads. Probably wouldn't come up but I think it's good to keep this check since it'll become more relevant once we allow evaluating sets of features (with tags or all features in a project)